# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/cs_clip_rectangle.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/cs_clip_rectangle.glsl
--- a/gfx/wr/webrender/res/cs_clip_rectangle.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/cs_clip_rectangle.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -4,21 +4,21 @@
 
 #include shared,clip_shared,ellipse
 
-varying vec4 vLocalPos;
+varying highp vec4 vLocalPos;
 #ifdef WR_FEATURE_FAST_PATH
-flat varying vec3 vClipParams;      // xy = box size, z = radius
+flat varying mediump vec3 vClipParams;      // xy = box size, z = radius
 #else
-flat varying vec4 vClipCenter_Radius_TL;
-flat varying vec4 vClipCenter_Radius_TR;
-flat varying vec4 vClipCenter_Radius_BL;
-flat varying vec4 vClipCenter_Radius_BR;
-flat varying vec3 vClipPlane_TL;
-flat varying vec3 vClipPlane_TR;
-flat varying vec3 vClipPlane_BL;
-flat varying vec3 vClipPlane_BR;
+flat varying highp vec4 vClipCenter_Radius_TL;
+flat varying highp vec4 vClipCenter_Radius_TR;
+flat varying highp vec4 vClipCenter_Radius_BL;
+flat varying highp vec4 vClipCenter_Radius_BR;
+flat varying highp vec3 vClipPlane_TL;
+flat varying highp vec3 vClipPlane_TR;
+flat varying highp vec3 vClipPlane_BL;
+flat varying highp vec3 vClipPlane_BR;
 #endif
 // Clip mode. Packed in to a vector to work around bug 1630356.
-flat varying vec2 vClipMode;
+flat varying mediump vec2 vClipMode;
 
 #ifdef WR_VERTEX_SHADER
 